app/src/store/bootstrap.ts

```diff
@@ -69,22 +70,67 @@ export const bootstrapCallInvites = createTypedAsyncThunk<
   { rejectValue: BootstrapCallInvitesRejectValue }
 >(
   bootstrapCallInvitesActionTypes.prefix,
-  async (_, { dispatch, rejectWithValue }) => {
-    const handleCallInvite = (callInvite: CallInvite) => {
-      const callInviteInfo = getCallInviteInfo(callInvite);
-      const id = nanoid();
-      callInviteMap.set(id, callInvite);
-      dispatch(
-        setCallInvite({
-          id,
-          info: callInviteInfo,
-          status: 'idle',
-        }),
+  async (_, { dispatch, getState, rejectWithValue }) => {
+    /**
+     * Handle an incoming, pending, call invite.
+     */
+    const handlePendingCallInvite = async (callInvite: TwilioCallInvite) => {
+      await dispatch(receiveCallInvite(callInvite));
+    };
+    voice.on(Voice.Event.CallInvite, handlePendingCallInvite);
+
+    /**
+     * Handle the settling of a pending call invite.
+     */
+    const handleSettledCallInvite = (callInvite: TwilioCallInvite) => {
+      const callSid = callInvite.getCallSid();
+
+      const callInviteEntities = getState().voice.call.callInvite.entities;
+      const callInviteEntity = Object.values(callInviteEntities).find(
+        (entity) => {
+          if (typeof entity === 'undefined') {
+            return false;
+          }
+          return entity.info.callSid === callSid;
```

Comment:
Thoughts on doing something like `entity && entity.info.callSid === callSid` to make this simple?
```ts
const callInviteEntity = Object.values(callInviteEntities).find((entity) => entity && entity.info.callSid === callSid);
```

Reply:
So I've had somewhat of a "whiplash" of opinion on functional vs imperative lately LOL. I like functional and one-liners are kind of fun to concoct, but I think that leaning on the algebraic type system and "eliminating error states" imperatively is just as safe, and if we need to slap some debug console logs in here, we don't need to deconstruct a one-liner to do so. This use-case is pretty mild, but I like to do it everywhere now.

Reply:
It's been a while since I heard algebraic system lol. That makes me wonder, how is this related? Am I not seeing something?

re: functional vs imperative paradigm, we can debate all we want but I don't think it'll help us in this project. There are pros and cons which do not matter as much in a small project like this. But I like us to keep in mind that imperative style of coding usually produces code that is too verbose and hard to understand. I'm also not sure the ease of adding debug logs in this instance is convincing. Functional programming can also be debugged easily.

I'm also having hard time understanding the thought process/reasoning. If you take a look at my comment in this [section](https://github.com/twilio/twilio-voice-react-native-app/pull/84#discussion_r1263024073) of your PR, they seem to be accomplishing very similar thing, but implemented differently.

I don't want to push too hard for a specific way of doing things. But my ask is to decide on one style and be consistent.

Reply:
Yeah you're right, please see the latest changes.